packages/datadog_flutter_plugin/example/lib/rum_screen.dart

```diff
@@ -27,6 +35,14 @@ class _RumScreenState extends State<RumScreen> {
     super.dispose();
   }
 
+  void _getCurrentSessionId() async {
+    _currentSessionId = await DatadogSdk.instance.rum?.getCurrentSessionId();
+
+    if (mounted) {
+      setState(() {});
+    }
```

Comment:
just out of curiosity: what this thing is doing?

Reply:
In order to let Flutter know that the State (the object) has changed, you have to call `setState` so it marks the attached Widget as dirty. Usually, you put the modification of the state in the callback, but it's not necessary. Because I modified `currentSessionId` I have to then call `setState` to re-render. But, since it's an async method, there's a possibility the Widget / State has been detached from the tree, so I have to check `mounted` first (which checks if the widget is in the tree) to avoid an exception.